crates/storage/codecs/derive/src/lib.rs

```diff
@@ -106,10 +111,42 @@ pub fn derive_arbitrary(args: TokenStream, input: TokenStream) -> TokenStream {
 #[proc_macro_attribute]
 pub fn add_arbitrary_tests(args: TokenStream, input: TokenStream) -> TokenStream {
     let ast = parse_macro_input!(input as DeriveInput);
-    let tests = arbitrary::maybe_generate_tests(args, &ast);
+
+    let tests =
+        arbitrary::maybe_generate_tests(args, &ast.ident, &format_ident!("{}Tests", ast.ident));
     quote! {
         #ast
         #tests
     }
     .into()
 }
+
+struct GenerateTestsInput {
+    args: TokenStream,
+    ty: syn::Type,
+    mod_name: syn::Ident,
+}
+
+impl Parse for GenerateTestsInput {
+    fn parse(input: ParseStream<'_>) -> Result<Self> {
+        input.parse::<Token![#]>()?;
+
+        let args;
+        bracketed!(args in input);
+
+        let args = args.parse::<proc_macro2::TokenStream>()?;
+        let ty = input.parse()?;
+
+        input.parse::<Token![,]>()?;
+        let mod_name = input.parse()?;
+
+        Ok(Self { args: args.into(), ty, mod_name })
+    }
+}
+
+#[proc_macro]
+pub fn generate_tests(input: TokenStream) -> TokenStream {
```

Comment:
this makes sense, and I'd actually prefer macros over macro attributes for this.

could we add a few docs here?